modules/publish/controllers/Main.php

```diff
@@ -406,21 +406,22 @@ public function runToggleFavouriteArticle(framework\Request $request)
             // Validate request parameters.
             if ($article_id === null)
             {
-                return $this->renderText('no article');
+                return $this->return400(framework\Context::getI18n()->__('Article ID not specified'));
             }
 
             if ($user_id === null)
             {
-                return $this->renderText('fail');
+                return $this->return400(framework\Context::getI18n()->__('User ID not specified'));
             }
 
             // Retrieve article and user from database, making sure they exist.
             $article = Articles::getTable()->selectById($article_id);
             $user = \thebuggenie\core\entities\User::getB2DBTable()->selectById($user_id);
 
-            if ($article === null || $user === null )
+            if ($article === null || $user === null)
```

Comment:
check object type, not `null` - check that the object is of the type you expect, not any of the types you don't expect ;)